mantidproject/mantid
# New auto-scale options for sliceviewer

**Description of work**

This feature aims to add a "quick" auto-scale to the data to help users quickly scale the colormap to their data. This only applies to slice viewer but could be added to instrument view, as well.

**Purpose of work**


I often watch users "fiddle" with the colormap limits for a long time in order to view their data in the range they desire. This is a helper dropdown box that should improve things over min/max auto-scaling in some cases.



**Summary of work**


1. Default Min/Max (as before)
2. 3(Sig) -> mean ± standard deviation
3. 1.5(IQR) -> 25th, 75th-quartile ± 1.5 x interquartile range
4. 1.5(MAD) -> median ± 1.5 x median absolute deviation

**Further detail of work**


This is done is such a way as future types can quickly be added.

**To test:**

Open your favorite or interesting dataset, switch to several different scale types to make sure it works without issue. The colormap should be "zoomed" in for various options. Also be sure to change to different norm types.

1. Min/Max
![image](https://github.com/mantidproject/mantid/assets/13754794/de241fd2-3e39-4dc8-83ef-09ea15d5fadc)

2. 3(Sig)
![image](https://github.com/mantidproject/mantid/assets/13754794/465429a7-094f-4ccb-adb1-ddb6a4675ba6)

3. 1.5(IQR) 
![image](https://github.com/mantidproject/mantid/assets/13754794/7a919fd8-8d91-4c9e-9dfb-a3f7a7512b4b)

5. 1.5(MAD) 
![image](https://github.com/mantidproject/mantid/assets/13754794/a6c6322d-0635-48ec-9113-b787a1ce405e)





---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
